tests/RestApiTestCase.php

```diff
@@ -0,0 +1,195 @@
+<?php
+namespace Give\Tests;
+
+use WP_REST_Request;
+use WP_REST_Response;
+use WP_REST_Server;
+use WP_Roles;
+use WP_UnitTest_Factory;
+
+/**
+ * Give Unit API Test Case
+ *
+ * Provides Give-specific setup/tear down/assert methods
+ * and helper functions to be used in API tests.
+ *
+ * @unreleased
+ */
+class RestApiTestCase extends TestCase
+{
+    /**
+     * Test REST Server
+     *
+     * @unreleased
+     *
+     * @var WP_REST_Server
+     */
+    protected $server;
+
+    /**
+     * Test user accounts for API authentication
+     *
+     * @unreleased
+     *
+     * @var int[] $users
+     */
+    protected static $users = [
+        'anonymous' => null,
+        'administrator' => null,
+        'editor' => null,
+        'author' => null,
+        'contributor' => null,
+        'subscriber' => null,
+    ];
+
+    /**
+     * Create users for API authentication
+     *
+     * @unreleased
+     *
+     * @return void
+     */
+    public static function wpSetUpBeforeClass(WP_UnitTest_Factory $factory)
+    {
+        self::$users = [
+            'anonymous' => 0,
+            'administrator' => $factory->user->create(['role' => 'administrator']),
+            'editor' => $factory->user->create(['role' => 'editor']),
+            'author' => $factory->user->create(['role' => 'author']),
+            'contributor' => $factory->user->create(['role' => 'contributor']),
+            'subscriber' => $factory->user->create(['role' => 'subscriber']),
```

Comment:
What's the reason for populating all these users everytime? Are we just anticipating testing for different permissions related to the uer role?

Reply:
Yeah. It's a more generic approach so the REST API can be tested in different scenarios.

Reply:
I would be inclined to expand on this part a little bit.  In GiveWP we have specific user roles like `give_manager`, `give_donor`, etc. that we might want to test for.  Maybe we just move some of this setup into a trait like `HasDefaultWordPressUsers` and  `HasDefaultGiveWPUsers` so we can opt into this or not.